Vhar 7276 paivakirjan visualisointi

rckopteri opened the pull request:
Ensimmäinen versio dynaamisesti tehtävästä työmaapäiväkirjan raportista ja listauksesta

rckopteri committed:
Lisää vahvuus-taulukkoon oikeat tiedot päivystäjistä ja työnjohtajista

rckopteri committed:
Muuta työmaapäiväkirjan säähän liittyvän taulukon nimi sääasemaksi

Koska aseman tiedot niputetaan yhteen ja näytetään aikojen perusteella

rckopteri committed:
Laita poikkeuksellisiin keliolosuhteisiin tietokannasta tiedot

rckopteri committed:
Ensimmäinen versio dynaamisista säätiedoista

rckopteri committed:
Apufunktioita päivämäärien käsittelyyn työmaapäiväkirjatyökalulle

rckopteri committed:
Merge branch 'develop' into VHAR-7276-paivakirjan-visualisointi

rckopteri committed:
Poista erillinen vahingot tiedosto

rckopteri committed:
Poista muut_huomiot työmaapäiväkirja raportin tiedosto

rckopteri committed:
Poista turhaksi jäänyt tiedosto

rckopteri committed:
Poista turhaksi jäänyt tiedosto

rckopteri committed:
Siirrä toimeksiannot pois työmaapäiväkirjan_tapahtuma -taulusta

rckopteri committed:
Uudista hieman työmaapäiväkirjan ui puolta

rckopteri committed:
Päiväkirjalistauksen ja päiväkirjan tietojen hakemiseen ensimmäinen tietokantalause

rckopteri committed:
Dynaamisen työmaapäiväkirjan ensimmäisen version hahmottelua

rckopteri committed:
Dynaamisen työmaapäiväkirjaraportin hahmottelua

rckopteri committed:
Ensimmäinen versio hallintapaneelin työmaapäiväkirjatyökalusta

rckopteri committed:
Korjaa tietokantahaku ymmärtämään urakka id

rckopteri committed:
Siivoa roikkuvat sulkeet

rckopteri committed:
Pieni refaktorointi työmaapäiväkirjan ui:lle

rckopteri committed:
Parannan raportin tilan käsittelyä ja kevyt refaktorointi

rckopteri committed:
Estä raportin latautuminen, jos sitä kutsutaan liian aikaisin

rckopteri committed:
Mäppää kalusto tehtäviin md5 hashin avulla.

Tähän voisi keksiä jonkun toimivammankin keinon. Kuten muokata tietokantarakennetta tms.

rckopteri committed:
Poista käyttämättömät tiedostot

rckopteri committed:
Korjaa työmaapäiväkirjan tallennus ja testit

rckopteri committed:
Piilota muutoshistoria yläheaderista

rckopteri committed:
Refaktorointia

solita-staskila commented on src/cljs/harja/ui/tyokalut/raportti_tyokalut.cljs:
```diff
@@ -17,42 +18,48 @@
   ;; Tekee 2 taulukkoa vierekkän
   [:div.flex-gridit
-   [:div
+   [:div {:style {:width "50%"}}
```
Tälle taitaa olla classi olemassa nimellä tyomaa-haku-suodatin, classin voisi uudelleen nimetä esim "puolita-leveys", ja koodiin vaihtaa _[:div.puolita-leveys_ ... yhdenmukaisuuden nimissä

rckopteri replied:
👍  Vaihdoin classin nimeksi -> "puolita-leveys" ja otin käyttöön.

solita-staskila commented on src/cljs/harja/ui/tyokalut/raportti_tyokalut.cljs:
```diff
@@ -84,39 +91,41 @@
-       otsikot-oikea rivit-oikea])]])
+   ;; Ei piirretä oikeaa elementtiä, jos sitä ei ole annettu.
+   (if otsikko-oikea
+     [:div {:style {:width "50%"}}
```
Sama homma

solita-staskila commented on src/cljs/harja/ui/tyokalut/raportti_tyokalut.cljs:
```diff
@@ -84,39 +91,41 @@
+         sivuttain-rullattava?
+         ensimmainen-sarake-sticky?
+         otsikot-oikea rivit-oikea])]
+     [:div {:style {:width "50%"}}])])
```
Täälläkin

solita-staskila commented on src/clj/harja/palvelin/raportointi/raportit/tyomaapaivakirja/kalusto.clj:
```diff
@@ -11,15 +12,18 @@
+                                {:otsikko "Lisäkalusto" :otsikkorivi-luokka "nakyma-otsikko" :sarakkeen-luokka "nakyma-valkoinen-solu" :leveys 0.25 :tyyppi :varillinen-teksti}
+                                {:otsikko "Toimenpide" :otsikkorivi-luokka "nakyma-otsikko" :sarakkeen-luokka "nakyma-valkoinen-solu" :leveys 0.7 :tyyppi :varillinen-teksti})
+                :rivit (mapv
+                         #(kalusto-rivi (str (harja.pvm/aika (:aloitus %)) " - " (harja.pvm/aika (:lopetus %)))
```
Pitäisikö pvm mielummin requirata?

rckopteri replied:
Mietin samaa, mutta tuossa oli vähemmän merkkejä. Mutta koska se vaivaa muitakin kuin minua, niin requirasin :)

solita-staskila commented on src/clj/harja/kyselyt/tyomaapaivakirja.sql:
```diff
@@ -9,6 +9,82 @@ SELECT
+-- name: hae-paivakirjalistaus
+SELECT t.id as tyomaapaivakirja_id, t.urakka_id, u.nimi as "urakka-nimi",
+       d::DATE as paivamaara, -- Otetaan generoinnin päivämäärä
+       -- Ihan hihasta vedetty tilan määritelmä.
```
Kun työmaapäiväkirja puuttuu, olisiko silti hyvä näyttää kyseisen urakan nimi listauksessa? Esim. 

```sql
       SELECT t.id as tyomaapaivakirja_id, t.urakka_id, 
       COALESCE(u.nimi, (SELECT nimi FROM urakka where id = :urakka-id)) as "urakka-nimi",
....
```

rckopteri replied:
Mä en oo yhtään varma, että tuo urakan nimi pysyy tuossa ui:ssa loppuun asti. Mutta, hyvä pointti.
Lisäsin urakan nimen, vaikka itse työmaapäiväkirjaa ei olisikaan.

solita-staskila commented on src/clj/harja/palvelin/raportointi/raportit/tyomaapaivakirja/kalusto.clj:
```diff
@@ -11,15 +12,18 @@
-                                {:otsikko "Toimenpide" :otsikkorivi-luokka "nakyma-otsikko" :sarakkeen-luokka "nakyma-valkoinen-solu" :leveys 1 :tyyppi :varillinen-teksti})
-                :rivit [(kalusto-rivi "00:00" "6" "-" "Pistehiekoitus (01:23 - 02:15), Suolaus (00:22 - 01:12)")
-                        (kalusto-rivi "01:00" "6" "-" "Pistehiekoitus (01:23 - 02:15), Suolaus (00:22 - 01:12)")]}]
+                                {:otsikko "Aikaväli" :otsikkorivi-luokka "nakyma-otsikko" :sarakkeen-luokka "vaalen-tumma-tausta" :leveys 0.15 :tyyppi :varillinen-teksti}
```
Rivien välitykset karkaa hieman figman spekseistä. Tässä on suurinpiirtein oikea leveys, alempien taulukkojen ensimmäinen sarake tulisi myös olla saman pituinen. Nämä on silmäilty suurinpiirtein oikein, näitä on hieman vaikea asettaa tässä mutta onnistuu jotenkin

```suggestion
                                {:otsikko "Aikaväli" :otsikkorivi-luokka "nakyma-otsikko" :sarakkeen-luokka "vaalen-tumma-tausta" :leveys 0.14 :tyyppi :varillinen-teksti}
```

rckopteri replied:
Muutin kaikki nämä leveydet

solita-staskila commented on src/clj/harja/palvelin/raportointi/raportit/tyomaapaivakirja/keliolosuhteet.clj:
```diff
@@ -10,13 +10,18 @@
+(defn poikkeukselliset-keliolosuhteet-taulukko [poikkeussaat]
   (let [tiedot {:gridin-otsikko "Omat havainnot"
                 :rivin-tiedot (rivi
                                 {:otsikko "Klo" :otsikkorivi-luokka "nakyma-otsikko" :sarakkeen-luokka "vaalen-tumma-tausta" :leveys 0.098 :tyyppi :varillinen-teksti}
```
Sarakkeen leveys figman speksin mukaiseksi
```suggestion
                                {:otsikko "Klo" :otsikkorivi-luokka "nakyma-otsikko" :sarakkeen-luokka "vaalen-tumma-tausta" :leveys 0.155 :tyyppi :varillinen-teksti}
```

solita-staskila commented on src/clj/harja/palvelin/raportointi/raportit/tyomaapaivakirja/muut_toimenpiteet.clj:
```diff
@@ -1,23 +1,24 @@
+                       (str/join (:toimenpiteet %)))
+                toimenpiteet)
+        tiedot {:rivin-tiedot (rivi
+                                {:otsikko "Aikaväli" :otsikkorivi-luokka "nakyma-otsikko" :sarakkeen-luokka "vaalen-tumma-tausta" :leveys 0.15 :tyyppi :varillinen-teksti}
```
Sarakkeen leveys figman speksin mukaiseksi
```suggestion
                                {:otsikko "Aikaväli" :otsikkorivi-luokka "nakyma-otsikko" :sarakkeen-luokka "vaalen-tumma-tausta" :leveys 0.116 :tyyppi :varillinen-teksti}
```

solita-staskila commented on src/clj/harja/palvelin/raportointi/raportit/tyomaapaivakirja/saatiedot.clj:
```diff
@@ -24,24 +33,51 @@
                            {:otsikko "S-olom" :leveys 1 :otsikkorivi-luokka "nakyma-otsikko" :sarakkeen-luokka "nakyma-valkoinen-solu" :tyyppi :varillinen-teksti}
```
Yritin laittaa suggestionin mutta github dev kaatuu tälle riville, tässä rivien leveydet olisi 
Klo: 1
Loput sarakkeet: 0.75

solita-staskila commented on src/cljs/harja/views/hallinta/tyomaapaivakirjatyokalu_nakyma.cljs:
```diff
@@ -0,0 +1,116 @@
+
+(defn simuloi-tyomaapaivakirja []
+  (do
+    (println "jees")
```
En usko että tätä printtiä tarvitsee enää

rckopteri replied:
👍

solita-staskila commented on src/cljs/harja/ui/tyokalut/raportti_tyokalut.cljs:
```diff
@@ -84,39 +91,41 @@
+     [:div {:style {:width "50%"}}])])
 
 (defmethod raportointi/muodosta-html :tyomaapaivakirjan-kommentit [[_]]
   ;; Työmaakokouksen laskutusyhteenvedon footer
```
Kommentti on minun jäljiltä jäänyt copy pastella, voisi poistaa 
```suggestion
```

rckopteri replied:
👍

solita-staskila commented on src/cljs/harja/views/urakka/tyomaapaivakirja/paivakirja.cljs:
```diff
@@ -64,17 +74,18 @@
-      [grid/grid {:tyhja "Ei Tietoja."
-                  :tunniste :id
+      [grid/grid {;:tyhja "Ei Tietoja."
+                  :tunniste :paivamaara
                   :voi-kumota? false
                   :piilota-toiminnot? true
-                  :jarjesta :id
+                  :jarjesta :paivamaara
                   :mahdollista-rivin-valinta? true
                   :rivin-luokka solu-fn
                   :rivi-klikattu #(e! (tiedot/->ValitseRivi %))}
```
Gridissä voisi ottaa paginaation käyttöön mikäli tuloksia tulee paljon

```suggestion
      [grid/grid {;:tyhja "Ei Tietoja."
                  :tunniste :paivamaara
                  :sivuta grid/vakiosivutus
                  :voi-kumota? false
                  :piilota-toiminnot? true
                  :jarjesta :paivamaara
                  :mahdollista-rivin-valinta? true
                  :rivin-luokka solu-fn
                  :rivi-klikattu #(e! (tiedot/->ValitseRivi %))}
```

rckopteri replied:
Otin käyttöön, mutta tuo menee aivan käyttökelvottomaksi, jos yrittää yli kuukauden mittaista ajanjaksoa hahmottaa.

solita-staskila commented on src/cljs/harja/views/urakka/tyomaapaivakirja/paivakirja.cljs:
```diff
@@ -14,45 +15,54 @@
+      [debug app {:otsikko "TUCK STATE"}]
       [:h1.header-yhteiset "Työmaapäiväkirja"]
 
       [:div.row.filtterit
```
Pitäisikö listauksen aikaväliä rajoittaa? Voiko query jäädä roikkumaan jos tuotannossa käyttäjä laittaa 100 vuotta aikaväliksi?

```clojure
[valinnat/aikavali tiedot/aikavali-atom {:otsikko "Aikaväli"
                                                :for-teksti "filtteri-aikavali"
                                                :luokka #{"label-ja-aikavali " "ei-tiukkaa-leveytta "}
                                                :ikoni-sisaan? true
                                                :vayla-tyyli? true}]
```

Tässä voi käyttää :aikavalin-rajoitus parametria, esimerkkinä views->urakka->laskutusyhteenveto.cljs rivi 211

rckopteri replied:
Hyvä pointti. Laitoin rajoituksen 6kk

solita-staskila commented on src/clj/harja/palvelin/raportointi/raportit/tyomaapaivakirja.clj:
```diff
@@ -1,45 +1,143 @@
+    (into ()
+      (conj
+        ;; Lisätään tyhjä rivi jos on tarpeen
+        (if (not (empty? tapahtumat))
```
Eikö tämä aja saman asian?


```clojure
(if (not (empty? tapahtumat))
->
(if (some? tapahtumat)

Tai voiko olla jopa:
(if tapahtumat
```

edit: 
Tai näin:
```clojure 
(if (empty? tapahtumat)
          (conj
            [:jakaja true]
            (yhteiset/placeholder-ei-tietoja ei-tapahtumia-teksti))
          tapahtumarivit)
```

Tuo `(if (not (empty? tapahtumat)) ` ei omaan silmään ainakaan näytä hyvältä, siinä on turha checkki

rckopteri replied:
Tapahtumat on tyhjä lista, joten (if tapahtumat) on aina true. Otin tuon ehdotuksesi (some? tapahtumat) käyttöön.

solita-staskila reviewed (changes requested):
Muutama risu mutta hyvälle näyttää

rckopteri committed:
Refaktorointia

rckopteri committed:
Parempi uudistettu hallintatyökalu työmaapäiväkirjojen testaamiseksi

rckopteri committed:
Mäppää työmaapäiväkirjan tehtävät ja kalustot yhteen

Toteutus on tehty hyvin lineaarisesti koodilla, mutta se olisi hyvä tehdä kokonaan sql:ssä.

rckopteri committed:
Korjaa työmaapäiväkirjalistauksen haku

rckopteri committed:
Päivitä vielä versionumero myös työmaapäiväkirjatauluun

rckopteri committed:
Hae ja päivitä versionumero työmaapäiväkirja taulusta

rckopteri committed:
Mahdollista kysely, jolla voi päätellä mikä on seuraava versionumero

rckopteri committed:
Rakenna virheenkäsittely versionumeroinnin ympärille

rckopteri committed:
Varmista, että tietokannan solmiutuminen ei estä raportin latautumista

rckopteri committed:
Lisää virhekoodeja ja niiden käsittelyä

rckopteri committed:
Lisää verisonumero raportille, kun muutoshistoriaa ei vielä ole

rckopteri committed:
Rakenna työmaapäiväkirjahallintatyökalulle automaattinen versionumeron hallinta

koska tilaa ei hallita mitenkään, niin tehdään se lennosta

rckopteri committed:
Merge branch 'develop' into VHAR-7276-paivakirjan-visualisointi

rckopteri committed:
Raportin ui välityksien korjailua katselmointikommenttien perusteella

rckopteri committed:
Refaktorointia katselmointikommenttien perusteella

jpluomajoki-solita commented on dev-resources/less/pages/tyomaapaivakirja.less:
```diff
@@ -56,7 +56,7 @@
   }
 
-  .tyomaa-haku-suodatin {
+  .puolita-leveys {
```
Tämä sopisi hyvin utility-first/koko.lessiin samaan tapaan kuin `width-full`

rckopteri replied:
Siirsin

jpluomajoki-solita commented on src/clj/harja/kyselyt/tyomaapaivakirja.sql:
```diff
@@ -9,12 +9,90 @@ SELECT
+
+-- name: hae-paivakirja
+SELECT t.id as tyomaapaivakirja_id, t.urakka_id, u.nimi as "urakka-nimi", t.versio, t.paivamaara::DATE,
+       -- Hihasta vedetty tilan määritelmä
```
Oliko tämä tarkentumassa myöhemmin vai tarkoitus varmistaa ennen tämän mergeä?

rckopteri replied:
Hyvä pointti. Löysin dokumenteista, että pitää olla 12h sisällä seuraavana päivänä käytössä.

jarnovayrynen commented on src/clj/harja/kyselyt/tyomaapaivakirja.sql:
```diff
@@ -9,12 +9,90 @@ SELECT
+        WHERE versio = :versio AND tyomaapaivakirja_id = :tyomaapaivakirja_id) as tapahtumat,
+       (SELECT array_agg(row(kuvaus, aika))
+        FROM tyomaapaivakirja_toimeksianto
+        WHERE versio = :versio AND tyomaapaivakirja_id = :tyomaapaivakirja_id) as toimeksiannot,
```
Jäin jo silloin koodikahveilla aikanaan miettimään mutta unohdin kommentoida: toimisiko nuo subselectit näin: 
AND tyomaapaivakirja_id = :tyomaapaivakirja_id --> 
AND tyomaapaivakirja_id = t.id

Omaan silmään se näyttäis hivenen siistimmältä ja vähemmän fragiililta.

rckopteri replied:
Näyttääpi toimivan. Vaihdan. 👍

jpluomajoki-solita replied:
Tässä näyttäisi myös toimivan left join + array_agg, esim.

array_agg(row(tp.aloitus, tp.lopetus, tp.nimi)) as paivystajat,
...
  FROM tyomaapaivakirja t
         JOIN urakka u ON t.urakka_id = u.id
LEFT JOIN tyomaapaivakirja_paivystaja tp on t.id = tp.tyomaapaivakirja_id
Ainakin yksittäinen rivi miellyttää enemmän omaa silmää, mutta en sitten osaa sanoa onko tällä jotain suorituskyky tms. vaikutuksia.

rckopteri replied:
LEFT JOINilla lähdin liikkeelle. Se tuotaa useita rivejä per työmaapäiväkirja. Eli jos haluaa yhden työmaapäiväkirjan, jolla  on 5 riviä kalustoja ja vaikka 10 riviä tapahtumia, niin kahden LEFT JOIN:nin käytöllä (kalustot, tapahtumat) saadaan 15 riviä, vaikka tuloksiin yrittäisikin array_agg(row tyyppisesti niputtaa ne vastaukset.

rckopteri committed:
Muuta tietokantahakua paremmaksi. Korvaa annettu parametri suoralla tauluviittauksella

jpluomajoki-solita commented on src/clj/harja/kyselyt/tyomaapaivakirja.sql:
```diff
@@ -9,12 +9,90 @@ SELECT
+           WHEN t.luotu BETWEEN t.paivamaara and t.paivamaara + interval '14 day' THEN 'ok'
+           WHEN t.luotu > t.paivamaara + interval '14 day' THEN 'myohassa'
+           END as tila,
+       (SELECT array_agg(row(aloitus, lopetus, nimi))
```
Tässä näyttäisi myös toimivan left join + array_agg, esim.

array_agg(row(tp.aloitus, tp.lopetus, tp.nimi)) as paivystajat,
...
  FROM tyomaapaivakirja t
         JOIN urakka u ON t.urakka_id = u.id
LEFT JOIN tyomaapaivakirja_paivystaja tp on t.id = tp.tyomaapaivakirja_id
Ainakin yksittäinen rivi miellyttää enemmän omaa silmää, mutta en sitten osaa sanoa onko tällä jotain suorituskyky tms. vaikutuksia.
(näköjään tuli sama kommentti useasti...)

jpluomajoki-solita commented on src/clj/harja/kyselyt/tyomaapaivakirja.sql:
```diff
@@ -30,7 +108,7 @@ INSERT INTO tyomaapaivakirja_tyonjohtaja (urakka_id, tyomaapaivakirja_id, versio
 -- name: lisaa-saatiedot<!
-INSERT INTO tyomaapaivakirja_saa (urakka_id, tyomaapaivakirja_id, versio, havaintoaika, aseman_tunniste,
+INSERT INTO tyomaapaivakirja_saaasema (urakka_id, tyomaapaivakirja_id, versio, havaintoaika, aseman_tunniste,
                                   aseman_tietojen_paivityshetki, ilman_lampotila, tien_lampotila,
```
Sisennyksen voisi siistiä

rckopteri replied:
Siistitty

jpluomajoki-solita commented on src/clj/harja/palvelin/integraatiot/api/tyokalut/kutsukasittely.clj:
```diff
@@ -231,6 +231,11 @@
   (tee-viallinen-kutsu-virhevastaus virheet (get otsikot "origin")))
 
+(defn kasittele-duplikaatti-tyomaapaivakirja [resurssi otsikot virheet]
+  (log/warn (format "Resurssin: %s kutsu ei palauttanut hakutuloksia: %s " resurssi virheet))
```
Vastaako tämä varoitusviesti virhettä?

jpluomajoki-solita commented on src/clj/harja/palvelin/integraatiot/api/tyomaapaivakirja.clj:
```diff
@@ -29,6 +31,16 @@
+  (validointi/tarkista-urakka-ja-kayttaja db (:urakka_id tiedot) kayttaja)
+  (let [hakuparametrit {:urakka_id (:urakka_id tiedot)
+                        :paivamaara (pvm-string->java-sql-date (:paivamaara tiedot))}
+        _ (println "hae-tyomaapaivakirjan-versiotiedot :: hakuparametrit" (pr-str hakuparametrit))
```
Näkyykö nämä printlinet lokeilla?

jpluomajoki-solita commented on src/clj/harja/palvelin/raportointi/raportit/tyomaapaivakirja/kalusto.clj:
```diff
@@ -11,15 +13,18 @@
+                :rivit (mapv
+                         #(kalusto-rivi (str (pvm/aika (:aloitus %)) " - " (pvm/aika (:lopetus %)))
+                            (:tyokoneiden_lkm %) (or (:lisakaluston_lkm %) "-") (str/join ", " (:tehtavat %))
+                            #_"Miten saadaan tällainen mäppäys: Pistehiekoitus (01:23 - 02:15), Suolaus (00:22 - 01:12)")
```
Onko tämä todo-kommentti vai mikä?

jpluomajoki-solita commented on src/clj/harja/palvelin/raportointi/raportit/tyomaapaivakirja/saatiedot.clj:
```diff
@@ -13,35 +13,71 @@
+(defn- sateen-olomuoto
+  "Sateen olomuoto saadaan numeroarvosta, josta päätellään sateen olomuoto"
+  [numero]
+  ;;TODO: tee päättely tänne ja palauta oikea ikoni
```
Oliko tämä tulossa myöhemmin? Onko tikettiä?

jpluomajoki-solita commented on src/cljs/harja/tiedot/hallinta/tyomaapaivakirjatyokalu_tiedot.cljs:
```diff
@@ -0,0 +1,329 @@
+
+(defn- generoi-tieston-toimenpiteet [montako paivamaara]
+  (let [anna-toimenpide (fn [indeksi]
+                          (let [tunti (* indeksi 1)]
```
let taitaa olla tarpeeton?

rckopteri replied:
Kyllä. Tässä versiossa. Mutta jätän sen silti, koska se ikäänkuin dokumentoi tuon nimeämättömän funktion samalla.

jpluomajoki-solita commented on src/cljs/harja/views/hallinta.cljs:
```diff
@@ -97,6 +98,15 @@
 
+   "Työmaapäiväkirjatyökalu"
+   :tyomaapaivakirjatyokalu
+   (when true #_ (and (istunto/ominaisuus-kaytossa? :toteumatyokalu)
```
Onko tämä tarkoitus näkyä aina? Jos on, koko whenin voisi ottaa pois.

rckopteri replied:
Ei ole. Minulla oli cljc tiedostojen kääntöjen kanssa ongelmaa joka näkyi siinä, ettei työkalu tullut näkyviin, niin testailin tämmöstä. Oli ihan turha jäänne.. Hyvä löytö

jpluomajoki-solita commented on src/cljs/harja/views/urakka/tyomaapaivakirja/paivakirja.cljs:
```diff
@@ -0,0 +1,219 @@
+     ;:kommentoidut "Kommentoidut (123)" Lisätään kommentoidut sitten, kun niitä voi kommentoida
+     }))
+
+#_(def toimituksen-tila [{:class "ok" :selitys "Ok"}
```
Tämä näyttää vähän kehityksenaikaiselta testailulta, voinee poistaa?

rckopteri replied:
Poistettu

jpluomajoki-solita commented on src/cljs/harja/views/urakka/tyomaapaivakirja/paivakirja.cljs:
```diff
@@ -0,0 +1,219 @@
+                       "myohassa" {:class "myohassa" :selitys "Myöhässä"}
+                       "puuttuu" {:class "puuttuu" :selitys "Puuttuu"}})
+
+#_(def valittu-hakumuoto (reaction-writable :kaikki))
```
tässä sama kuin edellinen

rckopteri replied:
Poistettu

jpluomajoki-solita commented on tietokanta/src/main/resources/db/migration/V1_1020__.sql:
```diff
@@ -0,0 +1,44 @@
+-- Muuta työmaapäiväkirjan säätietotaulun nimeä vastaan paremmin sitä, mitä sillä haetaan
+ALTER TABLE tyomaapaivakirja_saa RENAME TO tyomaapaivakirja_saaasema;
```
Ei mitään parannusehdotusta, mutta saaasema on todella ärsyttävä kirjoittaa :smile:

rckopteri replied:
👯

jpluomajoki-solita reviewed (approved):
Yleisesti ottaen näyttää omaan silmään hyvältä. Muutamia pieniä koodin siistimiseen liittyviä asioita, mutta mitään huolestuttavaa toiminnallisuutta ei pistänyt silmään koodia katselemalla tai testaamalla käyttäen hallintatyökalua.

jpluomajoki-solita reviewed (approved):
Yleisesti ottaen näyttää omaan silmään hyvältä. Muutamia pieniä koodin siistimiseen liittyviä asioita, mutta mitään huolestuttavaa toiminnallisuutta ei pistänyt silmään koodia katselemalla tai testaamalla käyttäen hallintatyökalua.

jarnovayrynen commented on src/clj/harja/palvelin/raportointi/raportit/tyomaapaivakirja/muut_toimenpiteet.clj:
```diff
@@ -1,23 +1,24 @@
-                        (muut-toimenpiteet-rivi "04:00" "-")]}]
+(defn muut-toimenpiteet-taulukko [toimenpiteet]
+  (let [rivit (mapv #(muut-toimenpiteet-rivi
+                       (str (harja.pvm/aika (:aloitus %)) " - " (harja.pvm/aika (:lopetus %)))
```
Näyttää tosi samanlaiselle kuin mitä harja.pvm/aikavali-kellonajan-kanssa tekee, olisi hyvä käyttää sitä.

rckopteri committed:
Uusi kellonajan formatointifunktio ja sen käyttöönotto työmaapäiväkirjaraportissa

jarnovayrynen commented on src/clj/harja/palvelin/integraatiot/api/tyomaapaivakirja.clj:
```diff
@@ -29,6 +31,16 @@
+  (validointi/tarkista-urakka-ja-kayttaja db (:urakka_id tiedot) kayttaja)
+  (let [hakuparametrit {:urakka_id (:urakka_id tiedot)
+                        :paivamaara (pvm-string->java-sql-date (:paivamaara tiedot))}
+        _ (println "hae-tyomaapaivakirjan-versiotiedot :: hakuparametrit" (pr-str hakuparametrit))
```
Onko tarkoituksella println, vai pitäskö siivoilla tai muuttaa esim log/debug?

rckopteri replied:
Vaihdettu log/debug

rckopteri committed:
Siirretty puolita-leveys koko.lessiin eri nimellä

rckopteri committed:
Työmaapäiväkirjan myöhässä status päivitetty dokumenttien mukaiseksi

rckopteri committed:
Siivottu koodia katselmointikommenttien perusteella

solita-staskila commented on src/cljs/harja/tiedot/tyomaapaivakirja_tiedot.cljs:
```diff
@@ -0,0 +1,126 @@
+(defrecord HaeTiedotEpaonnistui [vastaus])
+
+(defn suodata-rivit
+  "Annetaan parametrina tilan :valinnat josta luetaan hakumuoto"
```
Tämä taitaa olla minun kirjoittama, tälle ei anneta tilan :valinnat parametrina vaan annetaan koko tila joten funktion kuvaus on väärä. 
Kirjoita uusi kuvaus, esim:

```suggestion
  "Suodatetaan tulokset käyttäjän valitsemien suodattimien perusteella"
```

rckopteri replied:
Vaihdettu

solita-staskila commented on src/cljs/harja/views/urakka/tyomaapaivakirja/paivakirja.cljs:
```diff
@@ -0,0 +1,214 @@
+  (let [#_(js/console.log "tyomaapaivakirja-listaus ::  app: " (pr-str app))
+        ;; TODO
+        ;; Lisää tähän oikea toiminnallisuus mikäli toimitus puuttuu (tekee "puuttuu-tausta" tekee oranssin solun taustan)
+        ;; Tällä hetkellä :tila tulee tyomaapaivakirja.sql joka on randomisti generoitu
```
Tämä kommentti ei taida pitää enään paikkansa. Voi poistaa.

rckopteri replied:
Jes. hyvä bongaus. Poistettu

solita-staskila commented on src/cljs/harja/views/urakka/tyomaapaivakirja/paivakirja.cljs:
```diff
@@ -0,0 +1,214 @@
+    {:kaikki (str "Kaikki (" (count (:tiedot tiedot)) ")")
+     :myohastyneet (str "Myöhästyneet (" (count myohassa) ")")
+     :puuttuvat (str "Puuttuvat (" (count puuttuu) ")")
+     ;:kommentoidut "Kommentoidut (123)" Lisätään kommentoidut sitten, kun niitä voi kommentoida
```
Onko kommenteille mitään tikettiä? Tuo pitää ilmeisesti toteuttaa jossain vaiheessa.

rckopteri replied:
Joo. Siellä on sellainen viimeistelytiketti, johon olen koonnut asioita, joita tässä ei ole toteutettu.

solita-staskila commented on src/cljs/harja/views/urakka/tyomaapaivakirja/paivakirja.cljs:
```diff
@@ -0,0 +1,214 @@
+                            :valitse-fn #(e! (tiedot/->PaivitaHakumuoto %))}
+         valittu-hakumuoto]]]
+
+      [grid/grid {;:tyhja "Ei Tietoja."
+                  :tunniste :paivamaara
+                  :sivuta grid/vakiosivutus
+                  :voi-kumota? false
+                  :piilota-toiminnot? true
+                  :jarjesta :paivamaara
+                  :mahdollista-rivin-valinta? true
+                  :rivin-luokka solu-fn
+                  :rivi-klikattu #(e! (tiedot/->ValitseRivi %))}
```
Voiko gridi olla tyhjä vai tuleeko tuloksia aina? Luulen että aina tulee vähintään "puuttuva" tulos, siinä tapauksessa tämän kommentin voi poistaa. Mutta jos gridi voi olla tyhjä, jonkin arvon voisi myös antaa.

```suggestion
      [grid/grid {:tunniste :paivamaara
                  :sivuta grid/vakiosivutus
                  :voi-kumota? false
                  :piilota-toiminnot? true
                  :jarjesta :paivamaara
                  :mahdollista-rivin-valinta? true
                  :rivin-luokka solu-fn
                  :rivi-klikattu #(e! (tiedot/->ValitseRivi %))}
```

rckopteri replied:
Otin käyttöön tyhjän rivin tekstillä: "Työmaapäiväkirjoja ei ole valitulle aikavälille."

solita-staskila commented on src/cljs/harja/views/urakka/tyomaapaivakirja/paivakirja.cljs:
```diff
@@ -0,0 +1,214 @@
+      [muodosta-html (assoc-in tiedot [1 :tunniste] tiedot/raportti-avain)]]
+
+     ;; Sticky bar (Edellinen - Seuraava) Tallenna PDF
+     ;;TODO: Toteutetaan myöhemmin
```
Pitääkö tiketöidä?

rckopteri replied:
Tästäkin on maininta sillä viimeistelytiketillä

rckopteri committed:
Estä tyhjien päiväkirjarivien generointi tulevaisuuteen

rckopteri committed:
Muokkauksia katselmointikommenttien perusteella
